safehttp/response_writer.go

```diff
@@ -80,6 +80,21 @@ func (w ResponseWriter) Header() Header {
 
 // Dispatcher TODO
 type Dispatcher interface {
-	Write(rw http.ResponseWriter, resp Response) error
-	ExecuteTemplate(rw http.ResponseWriter, t Template, data interface{}) error
+	Write(h ResponseWriterHolder, resp Response) error
+	ExecuteTemplate(h ResponseWriterHolder, t Template, data interface{}) error
+}
+
+// ResponseWriterHolder holds an http.ResponseWriter until
+// a status code and a content type to be written is provided.
+type ResponseWriterHolder struct {
+	w http.ResponseWriter
+}
+
+// Release releases the held http.ResponseWriter given that a status code
+// and a content type is provided. The status code and the content type are
+// written to the response before returning the responsewriter.
+func (h ResponseWriterHolder) Release(statusCode int, contentType string) http.ResponseWriter {
```

Comment:
Why not a `safehttp.StatusCode`?

Reply:
I feel like this statusCode is in a gray area between `net/http` and `safehttp` since it is used in the dispatcher. But it's probably still better to have it as a `safehttp.StatusCode` as you say. :+1: